React v18 update

andys8 opened the pull request:
This is the suggested update path to react v18:

* Rendering into `#root` (solves Solves https://github.com/JordanMartinez/purescript-cookbook/issues/198)
* Using `createRoot` as recommented in react v18
* Rename component to generic name (`app`), for consistency across examples.

If this change is good to go, I can roll it out to all `ReactHooks` examples.

```
./BookReactHooks/
./ButtonsReactHooks/
./CardsReactHooks/
./CatGifsReactHooks/
./ClockReactHooks/
./ComponentsInputReactHooks/
./ComponentsMultiTypeReactHooks/
./ComponentsReactHooks/
./DragAndDropReactHooks/
./FileUploadReactHooks/
./FormsReactHooks/
./GroceriesReactHooks/
./HelloReactHooks/
./ImagePreviewsReactHooks/
./NumbersReactHooks/
./PositionsReactHooks/
./RoutingHashReactHooks/
./RoutingPushReactHooks/
./ShapesReactHooks/
./TextFieldsReactHooks/
./TicTacToeReactHooks/
./TimeReactHooks/
```

JordanMartinez commented:
This looks good, but the only issue is the override in the `packages.dhall` file. (See https://github.com/JordanMartinez/purescript-cookbook/blob/master/CONTRIBUTING.md#only-use-packages-in-the-latest-package-set-release)

I get that the changes needed are in `master` for that library, but I'd still prefer to wait until an official release is made.

Other than that, yeah, this seems fine to me.

andys8 commented:
Yes, the override will be dropped and replaced by package set update before merge. Converted to draft PR to make clear this won't be merged in its current state.

JordanMartinez commented:
> Yes, the override will be dropped and replaced by package set update before merge. Converted to draft PR to make clear this won't be merged in its current state.

Sounds good!

JordanMartinez commented:
Looks like we only need to update the `react` dependency in Try PureScript, update the package set here to remove the override and use the latest release, and then this is good to merge. Is that right?

andys8 commented:
Aaaaand... update all the other react examples. Currently only one of them is modified. This can be another pullrequest though, since the api is deprecated, but not broken and will still work.

The latest package set doesn't have the react update. With the next package set, we'll have it.

* https://github.com/purescript/package-sets/commit/baec24fec2ed9a649cae3d9ccdfa83eb59180e23
* https://github.com/purescript/package-sets/commit/b0fadc9f670bbf51123d1ceb67cae7431769f207

andys8 commented:
Updated the package set. I would be fine with merging this PR as is:

* Only updating "HelloReactHooks".
* Next PR: Format repository with `purs-tidy`
* Next PR: Update all other react examples.

The current and expected state of the non-updated react examples is to see these warnings. They're exactly what this PR solved, but only for "HelloReactHooks". The examples with warnings still work.

![image](https://user-images.githubusercontent.com/13085980/180770402-1a2b2fdf-e135-4db7-b462-e37a76691c11.png)

![image](https://user-images.githubusercontent.com/13085980/180770435-91d5a4a5-21c4-42ad-a470-2bf1049f4bee.png)

JordanMartinez commented:
Thanks!
